alpha/apps/kaltura/lib/myContentStorage.class.php

```diff
@@ -72,9 +72,17 @@ public static function dirForId ( $id )
 		return (intval($id / 1048576)).'/'.	(intval($id / 1024) % 1024).'/'.$id;
 	}
 */
-	public static function dirForId ( $int_id, $id , $file_name = NULL )
+	public static function dirForId ( $int_id, $id , $file_name = NULL, $externalStorageMode = false )
 	{
-		return (intval($int_id / 1000000)).'/'.	(intval($int_id / 1000) % 1000).'/'. ( $file_name !== NULL ? $file_name : $id ) ;
+		if ($externalStorageMode && $id)
+		{
+			$dir =  substr($id, -2) . '/' . substr($id, -4, 2);
```

Comment:
replace order of Id so it will be more intutive to get path from ID 1_abcdefgh will be at /ef/gh/